code/workspaces/web-app/src/components/welcome/VerifyEmail.spec.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { createShallow } from '@material-ui/core/test-utils';
+import VerifyEmail from './VerifyEmail';
+import getAuth from '../../auth/auth';
+
+jest.mock('../../auth/auth');
+getAuth.mockImplementation(() => ({
+  login: jest.fn(),
```

Comment:
I find it useful to name mock functions as the name appears in any snapshots produced. Makes it easier to know you are passing the right function to the right prop of any child components!

e.g. `login: jest.fn().mockName('login')`